doc/api/util.md

```diff
@@ -858,6 +858,104 @@ added: v8.0.0
 * {symbol} that can be used to declare custom promisified variants of functions,
 see [Custom promisified functions][].
 
+## util.report.getReport([err])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `err` {Object}
+* Returns: {string} Returns the diagnostics report as a string
+
+Generates a json-formatted diagnostic report summary of the running process.
+The report includes JavaScript and native stack traces, heap statistics,
+platform information, resource usage etc. 
+
+```js
+const { report } = require('util');
+const data = report.getReport();
+console.log(data);
+```
+Additional documentation on diagnostic report is available
+at [node-report documentation][]
+
+## util.report.setDiagnosticReportOptions([events, signal, filename, path, verbose]);
+<!-- YAML
+added: REPLACEME
+-->
+
+* `events` {array}
+* `signal` {string}
+* `filename` {string}
+* `path` {string}
+* `verbose` {boolean}
+
+Runtime configuration of diagnostic report data capture. Upon invocation
+of this API, the Runtime is reconfigured to generate report based on 
+the new input.
+
+The events array can contain one or more of:
+
+`exception`: generate a report on unhandled exceptions
+
+`fatalerror`: generate a report on internal fault
+(such as out of memory errors or native assertions)
+
+`signal`: generate a report in response to a signal raised on the process.
+
+The other input are interpreted as follows:
+
+`filename`: name of the file to which the report will be written.
+
+`path`: location at which the report will be generated.
+
+`verbose`: flag that controls additional verbose information on
+report generation.
```

Comment:
It’s unusual to explain options like this. We almost always document their behaviour in the options list below the function, unless we need to go into a lot of detail (e.g. more than one paragraph).

Reply:
ok, sure

Reply:
Done, thanks.